### ISSUE TITLE
Reports bugs but nice error message

### ISSUE THREAD

**Issue by glenndevenish** [cjoudrey/graphql-schema-linter#160]
I'm setting up a schema from an existing REST api schema, and every now and again I do something like defining something twice, or not renaming all the types to graphql scalars.

When this happens, graphql-schema-linter reports a bug, but also a nice error message explaining what is wrong.

Why is this being reported as a bug in the linter?

Examples that I have come across so far:

```
It looks like you may have hit a bug in graphql-schema-linter.

It would be super helpful if you could report this here: https://github.com/cjoudrey/graphql-schema-linter/issues/new

Error: Type "Guid" not found in document.
```
and
```
It looks like you may have hit a bug in graphql-schema-linter.

It would be super helpful if you could report this here: https://github.com/cjoudrey/graphql-schema-linter/issues/new

Error: Type "Currency" was defined more than once.
```

**Comment by cjoudrey** [comment]
Thanks for opening this issue @glenndevenish.

I agree with you, this isn't a great user experience.

Back when I originally implemented this, GraphQL.js (the library I parse the schema dump), did not provide machine-readable errors when the schema was invalid (like the example you posted above).

I took a look and now it appears like there might be a method [`validateSDL`](https://github.com/graphql/graphql-js/blob/8d7a5fd8013232d0f8664c7c431671a830f74329/src/validation/validate.js#L57) which we could use _before_ building the schema which could give us nice errors back and improve this user experience.

I'll take a look shortly. 😄

**Comment by cjoudrey** [comment]
I took a look and it appears like https://github.com/graphql/graphql-js/pull/1626 will fix this issue.

I need to wait until GraphQL.js gets a version bump so that we can start using this. 🤞 

Thanks again for the report.

**Comment by glenndevenish** [comment]
Sweet! Thanks for the quick replies.

**Comment by cjoudrey** [comment]
Will be fixed in next release by #163.
